vector/src/main/java/im/vector/app/features/location/LocationSharingService.kt

```diff
@@ -208,6 +179,34 @@ class LocationSharingService : VectorService(), LocationTracker.Callback {
         destroyMe()
     }
 
+    @Synchronized
+    private fun addRoomArgs(beaconEventId: String, roomArgs: RoomArgs) {
+        roomArgsMap[beaconEventId] = roomArgs
+    }
+
+    @Synchronized
+    private fun removeRoomArgs(roomId: String) {
+        val beaconIds = roomArgsMap
+                .filter { it.value.roomId == roomId }
+                .map { it.key }
+        beaconIds.forEach { roomArgsMap.remove(it) }
```

Comment:
Nit: I would simplify this like:

```
roomArgsMap
                .filter { it.value.roomId == roomId }
                .forEach { roomArgsMap.remove(it.key) }
```

Reply:
Thanks! Fix here: [519d43c](https://github.com/vector-im/element-android/pull/6356/commits/519d43ceb78fc23c7f899cdf0033c2eb4c1055d8)